momepy/utils.py

```diff
@@ -470,8 +470,10 @@ def limit_range(vals, rng):
     array
         limited array
     """
+
     vals = np.asarray(vals)
-    if len(vals) > 2:
+
+    if (len(vals) > 2) and (not np.isnan(vals).all()):
```

Comment:
Cany you assing `np.isnan(vals)` to a variable and check against it? Because we do isnan here and then on line 482 again, so we can do it once and then do `var.all()` and `var.any()`.

Reply:
Tried this and it appears no significant timing change.

Reply:
Declaring a variable for `var = np.isnan(vals)` does not seem to make a significant difference in timing, but it also doesn't hurt. Probably should go with variable declaration?